pkg/odo/commonflags/run_on.go

```diff
@@ -0,0 +1,66 @@
+package commonflags
+
+import (
+	"errors"
+	"flag"
+	"fmt"
+
+	"github.com/redhat-developer/odo/pkg/odo/cmdline"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+)
+
+const (
+	// RunOnFlagName is the name of the flag allowing user to specify target platform
+	RunOnFlagName = "run-on"
+	RunOnCluster  = "cluster"
+	RunOnPodman   = "podman"
+	RunOnDefault  = RunOnCluster
+)
+
+// UseRunOnFlag indicates that a command accepts the --run-on flag
+func UseRunOnFlag(cmd *cobra.Command) {
+	if cmd.Annotations == nil {
+		cmd.Annotations = map[string]string{}
+	}
+	cmd.Annotations["runOn"] = "true"
+}
+
+// AddRunOnFlag adds the --run-on output flag to all commands
+// We use "flag" in order to make this accessible throughtout ALL of odo, rather than the
+// above traditional "persistentflags" usage that does not make it a pointer within the 'pflag'
+// package
+func AddRunOnFlag() {
+	flag.CommandLine.String(RunOnFlagName, "", "Specify target platform, supported platforms: cluster(default), podman")
+	_ = pflag.CommandLine.MarkHidden(RunOnFlagName)
+}
+
+// CheckRunOnCommand checks if commands enabling run-on flag are used correctly
+func CheckRunOnCommand(cmd *cobra.Command) error {
+
+	// Get the needed values
+	runOnFlag := pflag.Lookup(RunOnFlagName)
+	hasFlagChanged := runOnFlag != nil && runOnFlag.Changed
+	runOn := cmd.Annotations["runOn"]
+
+	// Check the valid output
+	if hasFlagChanged && runOnFlag.Value.String() != RunOnPodman && runOnFlag.Value.String() != RunOnCluster {
+		return fmt.Errorf("%s is not a valid target platform for --run-on, please select either cluster (default) or podman", runOnFlag.Value.String())
```

Comment:
Just to make it clear what the expected values are.
Also, even though #6150 will be adding an experimental mode, meanwhile, I think we could make it clear that `podman` support is experimental:

```suggestion
		return fmt.Errorf("%s is not a valid target platform for --run-on, please select either 'cluster' (default) or 'podman' (experimental)", runOnFlag.Value.String())
```